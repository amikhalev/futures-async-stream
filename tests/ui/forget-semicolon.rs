#![feature(async_await, generators)]

use futures_async_stream::async_stream;

#[async_stream(item = ())]
async fn foo() {
    yield;
    Some(()) //~ ERROR mismatched types
}

fn main() {}
