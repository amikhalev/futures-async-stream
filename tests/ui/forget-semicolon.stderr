error[E0308]: mismatched types
 --> $DIR/forget-semicolon.rs:8:5
  |
8 |     Some(()) //~ ERROR mismatched types
  |     ^^^^^^^^- help: try adding a semicolon: `;`
  |     |
  |     expected (), found enum `std::option::Option`
  |
  = note: expected type `()`
             found type `std::option::Option<()>`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
